lib/salus/scanners/osv/maven_osv.rb

```diff
@@ -0,0 +1,152 @@
+require 'salus/scanners/osv/base'
+
+module Salus::Scanners::OSV
+  class MavenOSV < Base
+    class SemVersion < Gem::Version; end
+
+    EMPTY_STRING = "".freeze
+    DEFAULT_SOURCE = "https://osv.dev/list".freeze
+    DEFAULT_SEVERITY = "MODERATE".freeze
+    GITHUB_DATABASE_STRING = "Github Advisory Database".freeze
+    MAVEN_OSV_ADVISORY_URL = "https://osv-vulnerabilities.storage.googleapis.com/"\
+        "Maven/all.zip".freeze
+
+    def should_run?
+      @repository.pom_xml_present?
+    end
+
+    def run
+      dependencies = find_dependencies
+      if dependencies.empty?
+        err_msg = "MavenOSV: Failed to parse any dependencies from the project."
+        report_stderr(err_msg)
+        report_error(err_msg)
+        return
+      end
+
+      @osv_vulnerabilities ||= fetch_vulnerabilities(MAVEN_OSV_ADVISORY_URL)
+      if @osv_vulnerabilities.nil?
+        err_msg = "MavenOSV: No vulnerabilities found to compare."
+        report_stderr(err_msg)
+        report_error(err_msg)
+        return
+      end
+
+      # Fetch vulnerable dependencies.
+      # Dedupe and select Github Advisory over other sources if available.
+      results = []
```

Comment:
It would be great if we can extract a method out of this.

```
results = vulnerable_dependencies_for(dependencies)
....

private
def vulnerable_dependencies_for(dependencies)
  results = []
  grouped = match_vulnerable_dependencies(dependencies).group_by { |d| d[:ID] }
  ...
  results
end
```

Reply:
Yup makes sense, let me update that.